src/ray/raylet/lineage_cache.cc

```diff
@@ -207,6 +225,27 @@ void LineageCache::RemoveWaitingTask(const TaskID &task_id) {
   // one.
   entry->ResetStatus(GcsStatus_UNCOMMITTED_REMOTE);
   RAY_CHECK(lineage_.SetEntry(std::move(*entry)));
+
+  // Try to evict a task and its uncommitted lineage if the uncommitted lineage
+  // exceeds the maximum size.
+  // NOTE(swang): The number of entries in the uncommitted lineage also
+  // includes local tasks that haven't been committed yet, not just remote
+  // tasks, so this is an overestimate.
+  const auto uncommitted_lineage = GetUncommittedLineage(task_id);
+  if (uncommitted_lineage.GetEntries().size() > max_lineage_size_) {
```

Comment:
What goes wrong if `max_lineage_size_ = 0`?

Reply:
Hmm, I think that will basically just try to evict every single entry in the lineage, but the code itself should not break. The tradeoff is basically the lower `max_lineage_size_` is, the more often a node will request notifications from the GCS.